CorsixTH/Lua/graphics.lua

```diff
@@ -120,18 +120,25 @@ end
 --! Tries to load the font file given in the config file as unicode_font.
 --! If it is not found it tries to find one in the operating system.
 function Graphics:loadFontFile()
+  local lfs = require("lfs")
+  local function check(path) return path and lfs.attributes(path, "mode") == "file" end
   -- Load the Unicode font, if there is one specified.
-  local font_file = self.app.config.unicode_font
-  if not font_file then
-    -- Try a font which commonly comes with the operating system.
-    local windir = os.getenv("WINDIR")
-    if windir and windir ~= "" then
-      font_file = windir .. pathsep .. "Fonts" .. pathsep .. "ARIALUNI.TTF"
-    elseif self.app.os == "macos" then
-      font_file = "/Library/Fonts/Arial Unicode.ttf"
-    else
-      font_file = "/usr/share/fonts/truetype/arphic/uming.ttc"
-    end
+  local config_path = self.app.config.unicode_font
+  -- Try a font which commonly comes with the operating system.
+  local os_path, font_file
+  local windir = os.getenv("WINDIR")
+  if windir and windir ~= "" then
+    os_path = windir .. pathsep .. "Fonts" .. pathsep .. "ARIALUNI.TTF"
+  elseif self.app.os == "macos" then
+    os_path = "/Library/Fonts/Arial Unicode.ttf"
+  else
+    os_path = "/usr/share/fonts/truetype/arphic/uming.ttc"
+  end
+  if check(config_path) then font_file = config_path
+  elseif check(os_path) then font_file = os_path
+  else
+    print("Could not find a Unicode font file")
```

Comment:
We never explicitly told the user before about this. And not sure they need to. Maybe gamelog it instead.

Reply:
Agreed, changed it to only message when there is a value to the config path. Has to be print, the gamelog isn't available yet. Should this be a language string? In a dialog?

Reply:
Maybe now I see why you previously wanted gamelog in app. It could warrant a UIInformation then. And it would need a language string.

Reply:
Falling back to the OS fonts isn't new, the new part is doing so even if a font is chosen in the config and it's not valid, and warning the user in that case if we fall through.

If we're tweaking the messaging for the player (who happens to have their console open), I would print when config_path is specified and not found regardless of whether we succeed in a fallback or not.

`Configured unicode font not found, using '....' instead.`
or
`Configured unicode font not found, no fallback available.`

Reply:
Done, plus saved the setting so that a warning isn't repeated.